README.md

```diff
@@ -411,6 +413,62 @@ When reduced to a prefix list, it would look like this:
 
 Which is then turned into an ste_vec of hashes which can be directly queries against the index.
 
+##### rich_jsonb opts
+
+A rich_jsonb index supports a richer set of JSONB functions and operators than an ste_vec index.
+
+Leaf nodes of the indexed JSONB document are also stored inline in the index which means that subsets of the document (down to individual fields and array elements) can be selected and returned to the client, avoiding the need for the client application to fetch the entire JSONB document.
+
+###### Supported functions & operators
+
+`cs_binop_v1(jsonb, '->', integer)`
+
+Extracts n'th element of JSON array (array elements are indexed from zero, but negative integers count from the end).
+
+`cs_binop_v1(jsonb, '->', text)`
+
+Extracts JSON object field with the given key.
+
+`cs_binop_v1(jsonb, '->>', integer)`
+
+Extracts n'th element of JSON array, as text.
+
+`cs_binop_v1(jsonb, '->>', text)`
+
+Extracts JSON object field with the given key, as text.
+
+`cs_binop_v1(jsonb, '<', jsonb)`
```

Comment:
How would `cs_binop_v1` determine which index to use for a particular operator? For example, if it needed to choose between regular ORE and rich JSON's ORE, how would that work?

Or is `cs_binop_v1` only intended to be used with rich JSON? In that case could we add the index name to the func name (something like `cs_rich_jsonb_binop_v1`)?

Reply:
@CDThomas given that a function can be polymorphic - I _think_ that means `cs_binop_v1` will have multiple definitions. So if both left and right args are the encrypted jsonb type, then the ORE index defined for that particular type will be used, for example.